### ISSUE TITLE
False negatives for functional TypedDicts + postponed evaluations

### ISSUE THREAD

**Issue by brentyi** [microsoft/pyright#10612]
Hello!

This script fails at runtime, but doesn't raise any type errors:
```python
from __future__ import annotations
from typing import TypedDict

X = TypedDict("X", {"field": Integer})  # Fails at runtime, no type error.
Integer = int
```
It looks like `Integer` is treated as a stringified annotation. Is this expected behavior?

[Pyright Playground reproduction](https://pyright-play.net/?code=GYJw9gtgBA%2BjwFcAuCQFM5QJYQA5hCSgEMA7UsJYpLMUgZwChRIokBPXLUgc2zwJEAKpzQATACJYAxkkaMAGlAC8UEbnFTZACgBEC3QBooAb13AsaADZjdALigBJUkjQ80IAL4BKRs9fuICrYLkA)

Thanks!!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.405
